main.py

```diff
@@ -61,18 +66,21 @@ def run(driver, shoe_type, username, password, url, shoe_size, login_time=None,
     except Exception as e:
         LOGGER.exception("Failed to login: " + str(e))
         six.reraise(Exception, e, sys.exc_info()[2])
-
+    
     if skip_retry_login is False:   
         try:
             retry_login(driver=driver, username=username, password=password)
         except Exception as e:
             LOGGER.exception("Failed to retry login: " + str(e))
             six.reraise(Exception, e, sys.exc_info()[2])
-            
+     
     if release_time:
         LOGGER.info("Waiting until release time: " + release_time)
         pause.until(date_parser.parse(release_time))
 
+    skip_add_address = False
+    skip_select_shipping = False
+    skip_payment = False
```

Comment:
Do we need these vars declared outside the loop? Like if you set them in one iteration, don't they need to be unset before the next iteration (if the first iteration failed)?

Reply:
You are correct. I have reset the variables after num_retries_attempted gets incremented.